### PR TITLE
update to ArrayInterface v7

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
See the release notes for details.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Can a v7 compatible version be tagged? https://github.com/SciML/DiffEqFlux.jl/issues/814#issuecomment-1438451007

**Comment by prbzrg** [comment]
Related to #189

**Comment by jonniedie** [comment]
It looks like `StaticArrayInterface` isn't registered. I usually gitignore Manifests. Is there a way to have an unregistered dependency without a Manifest?

File: test/runtests.jl [hunk 1]
```diff
@@ -619,8 +619,8 @@ end
     @test convert(Cholesky{Float32,Matrix{Float32}}, chol).factors isa Matrix{Float32}
 
     # Issue #140
-    @test ComponentArrays.ArrayInterfaceCore.indices_do_not_alias(typeof(ca)) == true
-    @test ComponentArrays.ArrayInterfaceCore.instances_do_not_alias(typeof(ca)) == false
+    @test ComponentArrays.StaticArrayInterface.indices_do_not_alias(typeof(ca)) == true
+    @test ComponentArrays.StaticArrayInterface.instances_do_not_alias(typeof(ca)) == false
```
Comment (ChrisRackauckas) [inline test/runtests.jl:623]:
```suggestion
    @test ComponentArrays.ArrayInterface.indices_do_not_alias(typeof(ca)) == true
    @test ComponentArrays.ArrayInterface.instances_do_not_alias(typeof(ca)) == false
```

**Comment by ChrisRackauckas** [comment]
Done

**Pull request merged**